### PR TITLE
Implements httpbeast support.

### PR DESCRIPTION
**Pull request by dom96**

### CODE REVIEW

File: websocket/server.nim [hunk 1]
```diff
@@ -60,7 +60,7 @@ proc makeHandshakeResponse*(key, protocol: string): string =
     result.add("Sec-Websocket-Protocol: " & protocol & "\c\L")
   result.add "\c\L"
 
-proc verifyWebsocketRequest*(req: Request, protocol = ""):
+proc verifyWebsocketRequest*[T](req: Request | T, protocol = ""):
```
Comment (Araq) [inline websocket/server.nim]: This is dirtier than overloading ``verifyWebsocketRequest`` would be, afaict.

File: websocket/server.nim [hunk 2]
```diff
@@ -80,32 +80,53 @@ proc verifyWebsocketRequest*(req: Request, protocol = ""):
   ## After successful negotiation, you can immediately start sending/reading
   ## websocket frames.
 
-  # if req.headers.hasKey("sec-websocket-extensions"):
+  when req.headers is Option[HttpHeaders]:
```
Comment (Araq) [inline websocket/server.nim]: Hacky.

File: websocket/server.nim [hunk 3]
```diff
@@ -80,32 +80,53 @@ proc verifyWebsocketRequest*(req: Request, protocol = ""):
...
 
-  let msg = makeHandshakeResponse(req.headers["sec-websocket-key"], protocol)
-  await req.client.send(msg)
-
-  return (AsyncWebSocket(kind: SocketKind.Server,
-    sock: req.client, protocol: protocol), "")
\ No newline at end of file
+  let msg = makeHandshakeResponse(headers["sec-websocket-key"], protocol)
+
+  when req.client is SocketHandle:
```
Comment (Araq) [inline websocket/server.nim]: Hacky.

File: websocket/server.nim [hunk 4]
```diff
@@ -80,32 +80,53 @@ proc verifyWebsocketRequest*(req: Request, protocol = ""):
...
+    req.forget()
```
Comment (Araq) [inline websocket/server.nim]: ``req.forget()``? At least needs a comment explaining this.

**Comment by dom96** [comment]
You're right. On second thought I decided to do this a little differently. This way will allow for support of various other HTTP servers.

**Comment by dom96** [comment]
Rebased.

**Pull request merged**